daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -71,7 +71,10 @@ final class Engine {
     *
     * The resulting transaction is annotated with packages required to validate it.
     */
-  def submit(cmds: Commands): Result[Transaction.Transaction] = {
+  def submit(
+      cmds: Commands,
+      transactionSeed: Option[crypto.Hash] = None
```

Comment:
a more descriptive name would be good (nodeIdSeed? I don't know...) and we should document how it is used.

Reply:
add description in the comment above